File: source/MyExpenses/MyExpensesDataGenerator/DataModel/PurchaseOrderItem.cs
namespace MyExpensesDataGenerator
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Data.Entity.Spatial;

    [Table("Purchase.PurchaseOrderItem")]
    public partial class PurchaseOrderItem
    {
        public int Id { get; set; }

        public int PurchaseHistoryId { get; set; }

        public int ProductId { get; set; }

        [Required]
        [StringLength(50)]
        public string ProductName { get; set; }

        public double Price { get; set; }

        public int ItemsNumber { get; set; }

        [Column(TypeName = "datetime2")]
        public DateTime PurchaseDate { get; set; }

        public virtual Product Product { get; set; }

        public virtual PurchaseOrder PurchaseOrder { get; set; }
    }
}

File: source/MyExpenses.Web_Experience2/server/model/Purchase/PurchaseOrderItem.js
'use strict';

module.exports = function (sequelize, DataTypes) {
    return sequelize.define('PurchaseOrderItem', {
        id: {
            type: DataTypes.INTEGER,
            allowNull: false,
            primaryKey: true,
            autoIncrement: true
        },
        purchaseHistoryId: {
            type: DataTypes.INTEGER,
            allowNull: false,
            references: {
                model: 'PurchaseOrder',
                key: 'Id'
            }
        },
        productId: {
            type: DataTypes.INTEGER,
            allowNull: false,
            references: {
                model: 'Product',
                key: 'Id'
            }
        },
        productName: {
            type: DataTypes.STRING,
            allowNull: false
        },
        price: {
            type: DataTypes.FLOAT,
            allowNull: false
        },
        itemsNumber: {
            type: DataTypes.INTEGER,
            allowNull: false
        },
        purchaseDate: {
            type: DataTypes.DATE,
            allowNull: false
        }
    }, {
        tableName: 'PurchaseOrderItem',
        timestamps: false,
        schema: 'Purchase'
    });
};
